### PR TITLE
Test post water 1699

### PR DESCRIPTION
**Pull request by bob0bob**
Changed it so when you go under water to turn off reverb.   I turned off the Low Filter because it is not needed with Reverb is turned off.

### CODE REVIEW

File: jme3-examples/src/main/java/jme3test/water/TestPostWater.java [hunk 1]
```diff
@@ -277,14 +277,12 @@ public void simpleUpdate(float tpf) {
...
-            waves.setDryFilter(new LowPassFilter(0.5f, 0.1f));
+           waves.setReverbEnabled(false);
             uw = true;
         }
         if (!water.isUnderWater() && uw) {
             uw = false;
-            //waves.setReverbEnabled(false);
-            waves.setDryFilter(new LowPassFilter(1, 1f));
-            //waves.setDryFilter(new LowPassFilter(1,1f));
+            waves.setReverbEnabled(true);
```
Comment (capdevon) [inline jme3-examples/src/main/java/jme3test/water/TestPostWater.java:285]: Interesting, so it is no longer necessary to continuously change the 'dryFIlter'?

Comment (stephengold) [reply jme3-examples/src/main/java/jme3test/water/TestPostWater.java:285]: I tested it, and it works. @bob0bob care to comment?

**Comment by bob0bob** [comment]
Made some changes.  There is now a hotkey to have the dryfilter on/off.  So someone running the example and see the difference.  Also, add all the hotkeys to the screen so someone running the demo can see the other keys that can change the examples.

File: jme3-examples/src/main/java/jme3test/water/TestPostWater.java [hunk 2]
```diff
@@ -205,12 +218,18 @@ public void onAction(String name, boolean isPressed, float tpf) {
                         water.setReflectionMapSize(Math.max(water.getReflectionMapSize() / 2, 32));
                         System.out.println("Reflection map size : " + water.getReflectionMapSize());
                     }
+                    if (name.equals("drayFilter"))
```
Comment (stephengold) [inline jme3-examples/src/main/java/jme3test/water/TestPostWater.java:221]: There's a typo in this line.

Comment (bob0bob) [reply jme3-examples/src/main/java/jme3test/water/TestPostWater.java]: thanks

**Comment by stephengold** [comment]
Thank you for adding UI hints! I wish every test did this.

1. On my keyboard, the keys are labelled PgUp and PgDn, not "Page Up" and "Page Dwn". Please correct the UI hints.

2. I think the `useDryFilter` logic is incomplete, because the flag doesn't take effect unless/until the `Camera` passes through the water's surface. Please correct the logic.

3. The imports should be in alphabetical order, with "com.jme3.font" after "com.jme3.audio". Please correct the order.

4. The "com.jme3.font.BitmapFont" import is unused and should be deleted.

5. It's not necessary to explicitly initialize `guiFont`. `SimpleApplication` does that for you, so you can delete the assignment to `guiFont`.

6. The open brace `{` belongs on the same line as the `if`, like so:
```java
        if (name.equals("dryFilter")) {
```

7. There should be a space on either side of the subtraction operator, like so:
```java
        txt2.setLocalTranslation(x, cam.getHeight() - y, 0);
```

**Comment by stephengold** [comment]
@bob0bob are you still available to work on this?

**Comment by bob0bob** [comment]
I made the changes.   But no I'm not available anymore for changes.   JME doesn't work for me and my time.  There are too many hands and no real organization to what is going on.  You have about 3 - 4 people with HUGE different views on what it should be done and there appears to be no real MAJOR structure on new ideas and what is happening.

This style of networking is not for me,  I'm short on time and to be fighting between people is not something I want to do.  This is the first time I've come across this type of unstructured format.   

I say this in hopes that maybe ideas for the future helps.  It appears that it is more of an idea of just people having libraries that are not part of JME and people use these unsupported libraries  and this has been going on for years.

**Comment by stephengold** [comment]
I'm very sorry to hear that the project hasn't met your needs.

I agree JME is very unstructured. It wasn't always this way, and I don't understand when it changed or why. What specific structures do you think the project needs?

It's normal in any group for people to have different visions of what should be done, so JMonkeyEngine isn't unusual in this regard. Unfortunately, I haven't been involved in any other open-source projects of similar complexity, so I don't have any basis of comparison. Do other projects hide their differences from the public by discussing them only in private? Do they empower authorities to impose their visions onto others? I can only imagine...

I don't understand your fear of becoming involved in fights. As far as I know, there's aren't currently any fights going on in the project. Rather, I'd say our stakeholders are too apathetic, not paying enough attention to what's happening and what needs to happen next. If there's conflict brewing that I'm unaware of, please enlighten me!

**Comment by bob0bob** [comment]
I've only been involved in 3 open source projects not including this one and most of them are structured.  They figure out a future for the project and they vote on it and stand behind that.

I got a message from one of your main people saying that I should be "THANKFUL" because I'm new and no new person should be contributing that quickly and it should take years before they are able to get anything part of the project and should only be making "LIBRARIES" and then post it for other people.     

Not "Fear" just not enough free time to fight other things.   Like people that want the status quo on LWJGL3 because there are not testers for MAC to solve it 100% even though it is less than 10% of the users,  they do not want anything to move forward even it is a fix for Linux and Windows.     There are other issues and gotten several emails from regular users that email privately because a "certain" someone ends up turning it into a fight because if you don't code like them they fight.

Sometime with power certain people go crazy,  it should be more about direction not dictation.

I've really enjoyed your help and you are very good, just to bad some of the other people are not at it.

**Comment by bob0bob** [comment]
We didn't comment, are those changes good?

**Comment by pspeed42** [comment]
Not sure who was fighting.  I personally only remember suggesting alternate approaches and in the end those were just opinions anyway.

Also I don't think any of the core devs suggested you should be thankful.  I certainly didn't.  We accept contributions from new folks right away if they slide right in.  Tenure really doesn't matter.

You wrote your own custom engine for your game and have a lot of very specific experience and perspective... all of which is valuable.  We have a longer looking perspective and our own set of experiences.  Also valuable.  But if discussion is viewed as a fight then this is definitely not the right project for you.  I seriously wish you luck on your projects, not even a hint of sarcasm.  You are clearly very passionate and talented and I'm sure will do some cool things.  

It's too bad the JME relationship was not productive.  In the long run, it probably hurts all involved but that's the way it goes sometimes.

**Comment by stephengold** [comment]
> I got a message from one of your main people saying that I should be "THANKFUL" because I'm new and no new person should be contributing that quickly and it should take years before they are able to get anything part of the project and should only be making "LIBRARIES" and then post it for other people.

I'm sorry to hear that someone with an aura of authority discouraged you from contributing. It's true we prefer to see new features implemented as add-on libraries before incorporating them into the Engine. However, I for one evaluate ideas and PRs based on their technical merit, not the seniority of the contributor. Some people are able to contribute productively from day one, while others (alas!) remain ineffective after many years of effort.

Whenever I remember to do so, I thank people for their contributions, regardless of their actual value. I never ask or expect anyone to be thankful for the opportunity to contribute. I wonder who it was who told you to be thankful ...

> are those changes good?

I plan to review them today. Since you said you're unavailable for further code changes, I won't ask you for any.

**Comment by stephengold** [comment]
@bob0bob thank you for your contribution to JMonkeyEngine!

**Pull request merged**